Static typing, flake8 liniting, docstrings & minor improvements

WardPearce opened the pull request:
Greetings,
I had some free time & thought I'd help contribute to this useful little project.

### I've added
- Static typing, so intellisense can help users when using this package.
- Flake8 liniting for consistent formatting.
- Docstrings so people understand how the class method works.
- Minor improvements, like declaring the `CurrencySymbols` class correctly in `__init__.py`, Removing pointless code & ensuring provided currency is always uppercase.

arshadkazmi42 reviewed (changes requested):
Thank you for this.

Since we added to make `currency` always uppercase, can you add one test case for this?

Where the test case will pass currency in smaller case and expects correct result.

WardPearce commented:
> Thank you for this.
> 
> Since we added to make `currency` always uppercase, can you add one test case for this?
> 
> Where the test case will pass currency in smaller case and expects correct result.

Good catch, added this.

arshadkazmi42 reviewed (approved):
Thank you for all the cleanups.

Looks good to me. :+1:
